pandas/core/window/rolling.py

```diff
@@ -1110,6 +1110,29 @@ class Window(BaseWindow):
     2  2.958621
     3       NaN
     4       NaN
+
+    **on**
+
+    Rolling sum with a window length of 2 on specific columon.
+
+    >>> df = pd.DataFrame({'A' : [1, 3, 5, np.nan, 7],
+    ...                    'B' : [2, 4, np.nan, 6, 8]})
+
+    >>> df
+        A    B
+    0  1.0  2.0
+    1  3.0  4.0
+    2  5.0  NaN
+    3  NaN  6.0
+    4  7.0  8.0
+
+    >>> df.rolling(2, on='A').sum()
```

Comment:
Without `on='A'`, the result will be like this:
```
>>> df.rolling(2).sum()
     A     B
0  NaN   NaN
1  4.0   6.0
2  8.0   NaN
3  NaN   NaN
4  NaN  14.0
```
Do I need to add this result as comparison?

Reply:
I was thinking more something like
```python
In [62]: df = pd.DataFrame({
    ...:     'A': to_datetime(['2020-01-01', '2020-01-01', '2020-01-02']),
    ...:     'B': [1,2,3],
    ...:     },
    ...:     index=date_range('2020', periods=3))
```
in which if you do `rolling('D')`, then the values of `'b'` differ if you do `on='A'` (instead of the default, which uses the index)

Reply:
Oh, I see what that means. How about the following:

```
>>> df = pd.DataFrame({
...     'A': [pd.to_datetime('2020-01-01'),
...           pd.to_datetime('2020-01-01'),
...           pd.to_datetime('2020-01-02'),],
...     'B': [1, 2, 3], },
...     index=pd.date_range('2020', periods=3))

>>> df
                    A  B
2020-01-01 2020-01-01  1
2020-01-02 2020-01-01  2
2020-01-03 2020-01-02  3

>>> df['B'].rolling('2D').sum()  # to avoid warning when sum on 'A'
2020-01-01    1.0
2020-01-02    3.0
2020-01-03    5.0
Freq: D, Name: B, dtype: float64

>>> df.rolling('2D', on='A').sum()   # value of 'B' is differ from above
                    A    B
2020-01-01 2020-01-01  1.0
2020-01-02 2020-01-01  3.0
2020-01-03 2020-01-02  6.0
```

Reply:
> # value of 'B' differs from above

other than that, looks fine

Reply:
Thanks for the correction, my English skills are a little rusty.